MonoGame.Framework/Platform/Native/Platform.Interop.cs

```diff
@@ -0,0 +1,299 @@
+// MonoGame - Copyright (C) The MonoGame Team
+// This file is subject to the terms and conditions defined in
+// file 'LICENSE.txt', which is part of this source code package.
+
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Input;
+using System.Runtime.InteropServices;
+
+namespace MonoGame.Interop;
+
+internal class MGHandleAttribute : System.Attribute
+{
+}
+
+internal enum EventType : uint
+{
+    Quit,
+
+    WindowMoved,
+    WindowResized,
+    WindowGainedFocus,
+    WindowLostFocus,
+    WindowClose,
+
+    KeyDown,
+    KeyUp,
+    TextInput,
+
+    MouseMove,
+    MouseButtonDown,
+    MouseButtonUp,
+    MouseWheel,
+
+    ControllerAdded,
+    ControllerRemoved,
+    ControllerStateChange,
+
+    DropFile,
+    DropComplete,
+}
+
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct MGP_WindowEvent
+{
+    public nint Window;
+    public int Data1;
+    public int Data2;
+}
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct MGP_KeyEvent
+{
+    public nint Window;
+    public uint Character;
+    public Keys Key;
+}
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct MGP_MouseMoveEvent
+{
+    public nint Window;
+    public int X;
+    public int Y;
+}
+
+internal enum MouseButton
+{
+    Left,
+    Middle,
+    Right,
+    X1,
+    X2
+}
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct MGP_MouseButtonEvent
+{
+    public nint Window;
+    public MouseButton Button;
+}
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct MGP_MouseWheelEvent
+{
+    public nint Window;
+    public int Scroll;
+    public int ScrollH;
+}
+
+internal enum ControllerInput : int
+{
+    INVALID = -1,
+
+    A,
+    B,
+    X,
+    Y,
+    Back,
+    Guide,
+    Start,
+    LeftStick,
+    RightStick,
+    LeftShoulder,
+    RightShoulder,
+    DpadUp,
+    DpadDown,
+    DpadLeft,
+    DpadRight,
+    Misc1,
+    Paddle1,
+    Paddle2,
+    Paddle3,
+    Paddle4,
+    Touchpad,
+    LAST_BUTTON = Touchpad,
+
+    LeftStickX,
+    LeftStickY,
+    RightStickX,
+    RightStickY,
+    LeftTrigger,
+    RightTrigger,
+    LAST_TRIGGER = RightTrigger,
+}
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct MGP_ControllerEvent
+{
+    public int Id;
+
+    public ControllerInput Input;
+
+    /// <summary>
+    /// For a axis input this can range from -32768 to 32767.
+    /// For a button input the value is 0 or 1.
+    /// </summary>
+    public short Value;
+}
+
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct MGP_DropEvent
+{
+    public nint Window;
+    public nint File;
+}
+
+
+[StructLayout(LayoutKind.Explicit)]
+internal struct MGP_Event
+{
+    [FieldOffset(0)]
+    public EventType Type;
+
+    [FieldOffset(4)]
+    public ulong Timestamp;
```

Comment:
I'm using the 64bit timestamp from `SDL_GetTicks64()`  BUT it seems SDL returns 32bit time stamps in their events.  Need advice here @mrhelmut @harry-cpp .